code/modules/antagonists/eldritch_cult/knowledge/flesh_lore.dm

```diff
@@ -123,21 +119,12 @@
 	heretic_monster.set_owner(master)
 	return
 
-/datum/eldritch_knowledge/flesh_grasp/proc/check_ghouls(mob/user)
-	if(LAZYLEN(spooky_scaries) == 0)
-		return
-
-	for(var/spook in spooky_scaries)
-		if(!ishuman(spook))
-			LAZYREMOVE(spooky_scaries, spook)
-			continue
-		var/mob/living/carbon/human/ghoul = spook
-		if(ghoul.stat == DEAD)
-			ghoul?.mind.remove_antag_datum(/datum/antagonist/heretic_monster)
-			LAZYREMOVE(spooky_scaries, spook)
-			continue
 
-	listclearnulls(spooky_scaries)
+/datum/eldritch_knowledge/flesh_grasp/proc/remove_ghoul(datum/source)
+	var/mob/living/carbon/human/humie = source
+	spooky_scaries -= humie
+	humie.mind.remove_antag_datum(/datum/antagonist/heretic_monster)
+	UnregisterSignal(source,COMSIG_MOB_DEATH,.proc/remove_ghoul)
```

Comment:
```suggestion
	UnregisterSignal(source, COMSIG_MOB_DEATH)
```
Unregister signal doesn't take a third argument.
Please, pretty please, start using spaces to make your code more readable.